routes/tasks.js

```diff
@@ -3,13 +3,16 @@ const router = express.Router();
 const authenticate = require("../middlewares/authenticate");
 const tasks = require("../controllers/tasks");
 const { createTask, updateTask, updateSelfTask } = require("../middlewares/validators/tasks");
-const { authorizeUser } = require("../middlewares/authorization");
+const authorizeRoles = require("../middlewares/authorizeRoles");
+const {
+  ROLES: { APPOWNER, SUPERUSER },
+} = require("../constants/roles");
 
 router.get("/", tasks.fetchTasks);
 router.get("/self", authenticate, tasks.getSelfTasks);
-router.get("/overdue", authenticate, authorizeUser("superUser"), tasks.overdueTasks);
-router.post("/", authenticate, authorizeUser("appOwner"), createTask, tasks.addNewTask);
-router.patch("/:id", authenticate, authorizeUser("appOwner"), updateTask, tasks.updateTask);
+router.get("/overdue", authenticate, authorizeRoles([SUPERUSER]), tasks.overdueTasks);
+router.post("/", authenticate, authorizeRoles([APPOWNER]), createTask, tasks.addNewTask);
+router.patch("/:id", authenticate, authorizeRoles([APPOWNER]), updateTask, tasks.updateTask);
```

Comment:
```suggestion
router.post("/", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), createTask, tasks.addNewTask);
router.patch("/:id", authenticate, authorizeRoles([APPOWNER, SUPERUSER]), updateTask, tasks.updateTask);
```

Previously `appOwner` gave permission to both app_owner and super_user. 
Now it gives access to only the role mentioned.